core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -152,37 +193,50 @@ private InfrastructureProvisionResult provisionInfrastructure(InfrastructureConf
             return provisionResult;
         } catch (TestGridInfrastructureException e) {
             setTestPlanStatus(testPlan, Status.FAIL);
-            throw new TestPlanExecutorException(StringUtil
+            String msg = StringUtil
                     .concatStrings("Error on infrastructure creation for deployment pattern '",
-                            testPlan.getDeploymentPattern(), "', in TestPlan"), e);
+                            testPlan.getDeploymentPattern(), "', in TestPlan");
+            logger.error(msg, e);
         } catch (InfrastructureProviderInitializationException | UnsupportedProviderException e) {
             setTestPlanStatus(testPlan, Status.FAIL);
-            throw new TestPlanExecutorException(StringUtil
+            String msg = StringUtil
                     .concatStrings("No Infrastructure Provider implementation for deployment pattern '",
-                            testPlan.getDeploymentPattern(), "', in TestPlan"), e);
+                            testPlan.getDeploymentPattern(), "', in TestPlan");
+            logger.error(msg, e);
+        } catch (Exception e) {
+            // We need to catch the Exception here since we need to catch and handle all exceptions within this method.
+            // This is required to do gracefully close any opened resources. Created wso2/testgrid#515 to track.
```

Comment:
I think we can remove this comment, as it is the issue being addressed by this PR :)